src/Password/Argon2i.php

```diff
@@ -0,0 +1,195 @@
+use Traversable;
+use Zend\Stdlib\ArrayUtils;
+
+use const PASSWORD_ARGON2I;
+use const PHP_VERSION_ID;
+
+use function is_array;
+use function password_hash;
+use function password_verify;
+use function strtolower;
+
+/**
+ * Argon2i algorithm using password_hash() function of PHP
+ */
+class Argon2i implements PasswordInterface
+{
+    /**
+     * Maximum memory (in kibibytes) that may be used to compute the Argon2 hash
+     *
+     * @var int|null
+     */
+    protected $memoryCost;
+
+    /**
+     * Maximum amount of time it may take to compute the Argon2 hash
+     *
+     * @var int|null
+     */
+    protected $timeCost;
+
+    /**
+     * Number of threads to use for computing the Argon2 hash
+     *
+     * @var int|null
+     */
+    protected $threads;
+
+    /**
+     * Constructor
+     *
+     * @param array|Traversable $options
+     *
+     * @throws Exception\InvalidArgumentException
+     */
+    public function __construct($options = [])
+    {
+        if (PHP_VERSION_ID < 70200) {
+            throw new Exception\InvalidArgumentException(
+                'The Argon2i password hash is only nativly available on PHP7.2+'
+            );
+        }
+
+        if (empty($options)) {
+            return;
+        }
+
+        if ($options instanceof Traversable) {
+            $options = ArrayUtils::iteratorToArray($options);
+        }
+
+        if (! is_array($options)) {
+            throw new Exception\InvalidArgumentException(
+                'The options parameter must be an array or a Traversable'
+            );
+        }
+
+        foreach ($options as $key => $value) {
+            switch (strtolower($key)) {
+                case 'memory_cost':
+                    $this->setMemoryCost($value);
+                    break;
+                case 'time_cost':
+                    $this->setTimeCost($value);
+                    break;
+                case 'threads':
+                    $this->setThreads($value);
+                    break;
+            }
+        }
+    }
+
+    /**
+     * Returns the maximum memory (in kibibytes) that may be used to compute the Argon2 hash
+     *
+     * @return int
+     */
+    public function getMemoryCost()
+    {
+        return $this->memoryCost;
+    }
+
+    /**
+     * Sets the maximum memory (in kibibytes) that may be used to compute the Argon2 hash
+     *
+     * @param int $memoryCost
+     *
+     * @return Argon2i
+     */
+    public function setMemoryCost($memoryCost)
+    {
+        $this->memoryCost = $memoryCost;
+        return $this;
+    }
+
+    /**
+     * Returns the maximum amount of time it may take to compute the Argon2 hash
+     *
+     * @return int
+     */
+    public function getTimeCost()
+    {
+        return $this->timeCost;
+    }
+
+    /**
+     * Sets the maximum amount of time it may take to compute the Argon2 hash
+     *
+     * @param int $timeCost
+     *
+     * @return Argon2i
+     */
+    public function setTimeCost($timeCost)
```

Comment:
We should check the value of `$timeCost` and verify that is allowed values for `$options` in `password_hash()`. As commented before, we should offer a safe way to use the Argon2i parameter with a detailed message in case of invalid values.